admin/server/auth/auth.go

```diff
@@ -64,3 +64,9 @@ func NewAuthenticator(logger *zap.Logger, adm *admin.Service, cookieStore *cooki
 
 	return a, nil
 }
+
+func NewMockAuthenticator(adminService *admin.Service) *Authenticator {
+	return &Authenticator{
+		admin: adminService,
+	}
+}
```

Comment:
`NewAuthenticator` actually creates 
```
oidcProvider, err := oidc.NewProvider(context.Background(), "https://"+opts.AuthDomain+"/")
```
which will fail so thats why create a new mock one

Reply:
Don't we do the same thing with mocking libraries ? We mock the object and return values of methods on that object that we expect to be called. Upon adding new tests or change in existing flow, tests break and then we add new methods that are expected to be called now.

Reply:
I tried `gorillio-stage.auth0.com` as AuthDomain and it worked, was not sure if tests would have internet access.

Reply:
Yes it's expected with mocks. But in this case, it's not really a mock authenticator (responding with mocked values), but a real one that will just fail on certain code paths. And we actually want to test parts of this authenticator – so it would be nicer if we got the "real" one from NewAuthenticator.

Generally, if testing something that has code paths that shouldn't run in tests (in this case, the oidc provider stuff), we need to then figure out a way to prevent that – might involve refactoring, to decouple the parts we want to test from those that should be replaced with a mock dependency (e.g. by introducing interfaces and using dependency injection like in the Github interface case). 

In this case, if it seems to work by passing the stage auth domain, it's probably fine for now if you don't want to spend more time on it.